research/audioset/yamnet/features.py

```diff
@@ -77,3 +76,18 @@ def spectrogram_to_patches(spectrogram, params):
     # features has shape [<# patches>, <# STFT frames in an patch>, MEL_BANDS]
 
     return features
+
+
+def pad_waveform(waveform, params):
+  """Pads waveform with silence if needed to get at least one input patch."""
+  # In order to produce one patch of log mel spectrogram input to YAMNet, we
+  # need at least one patch window length of waveform plus enough extra samples
+  # to complete the final STFT analysis window.
+  min_waveform_seconds = (params.PATCH_WINDOW_SECONDS + params.STFT_WINDOW_SECONDS
+                          - params.STFT_HOP_SECONDS)
```

Comment:
I hate the way the line break gathers PATCH_WINDOW_SECONDS and STFT_WINDOW_SECONDS.  It's STFT_WINDOW_SECONDS and STFT_HOP_SECONDS that should be gathered.  But again, no biggie.

Reply:
Rewrapped. And while I was in here, I made the padding more general and rounded up to the next hop as well.